### DISCUSSION TITLE
Do iPXE support Wireless NIC "Wi-Fi 6E AX211" ?

### DISCUSSION THREAD

**Discussion by weitheng10** [ipxe/ipxe#959] [Q&A]
After PXE boot with ipxe 1.21.1 i facing error code "https://ipxe.org/err/1a0861". 

I look up from "https://ipxe.org/appnote/hardware_drivers" dont find the devices id "8086:2725"  therefore not sure sure do ipxe support this nic.

I look up the device id from here "https://pci-ids.ucw.cz/read/PC/8086/2725"

im using snp.efi for my nbp file
![image](https://github.com/ipxe/ipxe/assets/95264824/9e18f114-aee2-42fd-8096-4c615e9781de)

**Comment by NiKiZe** [comment]
Unless your bios/firmware provides a SNP or NII interface for the NICthen no.
You should grab pciid with a pciid tool or similar.
You should use ifstat to show the nics you have in iPXE, and which driver they are using.

Most wifi nics do require binary blobs, so that makes them unlikely to get any support.
Recommendation here is to use a mobile device in thetering mode and iPXE build with USB support.

**Reply by weitheng10** [reply]
ifstat results:
![image](https://github.com/ipxe/ipxe/assets/95264824/0ac6b1fd-2324-4bcf-b230-a85ec1760ebe)

"Most wifi nics do require binary blobs, so that makes them unlikely to get any support." -> could you share more about this I dont quite understand 

"Recommendation here is to use a mobile device in thetering mode and iPXE build with USB support."  -> do u mean by using a usb adapter instead of wireless connection ?

If i would like to PoC on including the Wi-Fi driver, is there any reference on how to do it?

As of what i know there are 2 things to be handle (guidance needed)
1) How to identify what NIC driver to use. (I have found the NIC model but dono where to get the driver)
2) how to include the driver into iPXE and rebuild the efi

please to let me know if any other requirement .

**Reply by NiKiZe** [reply]
Drivers in iPXE are written from scratch in C based on hardware reference manuals that you need to get from the manufacturer (some times under NDA)
So there is no "where to get the driver".
